### DISCUSSION TITLE
How to get Huffman code for each node?

### DISCUSSION THREAD

**Discussion by graber1** [mapequation/infomap#337] [Q&A] [answered]
Hi, can you tell me how to get Huffman code for each of the nodes?

**Comment by antoneri** [comment, marked as answer]
Hi!

Infomap doesn't need to generate Huffman codes in practice, instead we calculate the stationary visit-rates.
Using those, we can calculate the lower bound for the expected modular codelength.

We call the visit-rates "flow", and you can find those values in the second column of the tree file.
[Check here for more info about the format](https://www.mapequation.org/infomap/#OutputTree)

If you want to assign Huffman codes, you can use those values and put it into a Huffman code generator or implement it yourself.

We have [a visualization](https://www.mapequation.org/demo/) of what a modular Huffman coding scheme would look like, but as I said, we don't need to do that in practice.

Hope this helps!
